src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -201,4 +202,22 @@ public static SuggestionType parseSuggestionType(String suggestionType) throws P
         }
         return suggestionSet;
     }
+
+    /**
+     * Parses a {@code String dateString} into a {@code ContractExpiryDate}.
+     */
+    public static ContractExpiryDate parseContractExpiryDate(String dateString) throws ParseException {
+        requireNonNull(dateString);
+        String trimmedDateString = dateString.trim();
+        if (!ContractExpiryDate.isValidDate(trimmedDateString)) {
+            throw new ParseException(ContractExpiryDate.MESSAGE_CONSTRAINTS);
+        }
+        LocalDate date;
+        try {
+            date = LocalDate.parse(trimmedDateString, ContractExpiryDate.DATE_FORMATTER);
+        } catch (Exception e) {
+            throw new ParseException(ContractExpiryDate.MESSAGE_CONSTRAINTS);
+        }
```

Comment:
Should this be used instead to check for validity of a date string? Then we could do away with the regex since it doesn't actually filter out 31-Feb.

Reply:
I'm not sure if and how the LocalDate parser does its parsing though, so I would prefer to be safer by mandating the regex check. For example, it might parse 20/19/020 as a correct date and not throw any errors. Apparently using the ResolverStyle.STRICT is supposed to prevent this kind of mis-parsing, but I think the regex would be more defensive coding-like